packages/core/src/card-model.ts

```diff
@@ -249,46 +253,28 @@ export default abstract class CardModel implements CardModelInterface {
     done!();
   }
 
-  protected async beginRecompute(): Promise<void> {
-    // This is a hook for subclasses to override if there is initial async work
-    // to do before doing the recompute
-  }
-
-  // we have a few places that are very sensitive to the shape of the data, and
-  // won't be able to deal with a schema instance that has additional properties
-  // and methods beyond just the data itself, so this method is for those places
-  private shapeData(shape: 'used-fields' | 'all-fields') {
-    let syncData: Record<string, any> = {};
-
-    let fields = shape === 'used-fields' ? this.usedFields : this.allFields;
-    for (let field of fields) {
-      let value = get(this._schemaInstance, field);
-      // undefined is a signal that the field should not exist
-      if (value !== undefined) {
-        set(syncData, field, value);
-      }
-    }
-    return syncData;
+  private schemaSerialize(
+    action: 'serialize' | 'deserialize',
+    format: Format | 'all',
+    data?: Record<string, any>
+  ): Record<string, any> {
+    return this._schemaInstance[this.schemaModule.serializeMember](action, format, data);
   }
 
-  private async createSchemaInstance() {
-    let klass = await this.schemaClass();
+  private createSchemaInstance() {
+    let klass = this.schemaModule.default;
     // We can't await the instance creation in a separate, as it's thenable and confuses async methods
     return new klass(this.getRawField.bind(this)) as any;
   }
 
   private getRawField(fieldPath: string): any {
-    let value = get(this.rawData, fieldPath);
-    return serializeField(this.serializerMap, fieldPath, value, 'deserialize');
-  }
-
-  private async schemaClass() {
-    if (this._schemaClass) {
-      return this._schemaClass;
+    let result = keySensitiveGet(this.rawData, fieldPath);
+    if ('missing' in result) {
+      throw new Error(`TODO: ${result.missing}`);
+    } else {
+      // TODO it would be wonderful if the schema instance knew how to deserialize its own fields
+      return serializeField(this.schemaModule.serializerMap, fieldPath, result.value, 'deserialize');
```

Comment:
This is the last instance of the `CardModel` directly handling serialization. Perhaps we could push this into the schema instance? this is a little different in that it is deserializing a specific field rather than all the fields for the format. Maybe that means pushing this whole method into the schema instance?